pkg/kubelet/client/kubelet_client.go

```diff
@@ -194,7 +195,7 @@ func (k *NodeConnectionInfoGetter) GetConnectionInfo(ctx context.Context, nodeNa
 	// Find a kubelet-reported address, using preferred address type
 	host, err := nodeutil.GetPreferredNodeAddress(node, k.preferredAddressTypes)
 	if err != nil {
-		return nil, err
+		return nil, errors.NewBadRequest(err.Error())
```

Comment:
this is actually an internal error, not the requester's fault... I could see a 503 instead of a 500